spec.html

```diff
@@ -35015,42 +34956,91 @@ <h1>get %TypedArray%.prototype [ @@toStringTag ]</h1>
         <li>has a *"length"* property whose value is 3.</li>
       </ul>
 
-      <emu-clause id="sec-typedarray">
-        <h1>_TypedArray_ ( )</h1>
-        <p>This description applies only if the _TypedArray_ function is called with no arguments.</p>
+      <emu-clause id="sec-typedarray" oldids="sec-typedarray-length">
+        <h1>_TypedArray_ ( ..._args_ )</h1>
+        <p>Each _TypedArray_ constructor performs the following steps when called:</p>
         <emu-alg>
           1. If NewTarget is *undefined*, throw a *TypeError* exception.
           1. Let _constructorName_ be the String value of the Constructor Name value specified in <emu-xref href="#table-the-typedarray-constructors"></emu-xref> for this <var>TypedArray</var> constructor.
-          1. Return ? AllocateTypedArray(_constructorName_, NewTarget, <code>"%<var>TypedArray</var>.prototype%"</code>, 0).
-        </emu-alg>
-      </emu-clause>
-
-      <emu-clause id="sec-typedarray-length">
-        <h1>_TypedArray_ ( _length_ )</h1>
-        <p>This description applies only if the _TypedArray_ function is called with at least one argument and the Type of the first argument is not Object.</p>
-        <p>_TypedArray_ called with argument _length_ performs the following steps:</p>
-        <emu-alg>
-          1. Assert: Type(_length_) is not Object.
-          1. If NewTarget is *undefined*, throw a *TypeError* exception.
-          1. Let _elementLength_ be ? ToIndex(_length_).
-          1. Let _constructorName_ be the String value of the Constructor Name value specified in <emu-xref href="#table-the-typedarray-constructors"></emu-xref> for this <var>TypedArray</var> constructor.
-          1. Return ? AllocateTypedArray(_constructorName_, NewTarget, <code>"%<var>TypedArray</var>.prototype%"</code>, _elementLength_).
+          1. Let _proto_ be <code>"%<var>TypedArray</var>.prototype%"</code>.
+          1. Let _numberOfArgs_ be the number of elements in _args_.
+          1. If _numberOfArgs_ = 0, then
+            1. Return ? AllocateTypedArray(_constructorName_, NewTarget, _proto_, 0).
+          1. Else,
+            1. Let _firstArgument_ be _args_[0].
+            1. If Type(_firstArgument_) is Object, then
+              1. Let _O_ be ? AllocateTypedArray(_constructorName_, NewTarget, _proto_).
+              1. If _firstArgument_ has a [[TypedArrayName]] internal slot, then
+                1. Perform ? InitializeTypedArrayFromTypedArray(_O_, _firstArgument_).
+              1. Else if _firstArgument_ has an [[ArrayBufferData]] internal slot, then
+                1. If _numberOfArgs_ &gt; 1, let _byteOffset_ be _args_[1]; else let _byteOffset_ be *undefined*.
+                1. If _numberOfArgs_ &gt; 2, let _length_ be _args_[2]; else let _length_ be *undefined*.
```

Comment:
is there defined behavior for accessing a List out-of-bounds? meaning, can `args[1]` just naturally be `undefined` when the list lacks that item?

Reply:
There is not, and I'd be opposed to adding such a thing.

Reply:
> is there defined behavior for accessing a List out-of-bounds?

Not currently.

> meaning, can args[1] just naturally be `undefined` when the list lacks that item?

It *could*, but I think I prefer it being a spec bug. If, say, a spec author makes an off-by-one error and accesses past the end of a List, I'd prefer the confidence of knowing that that has no defined meaning, rather than wondering if it's an intentional use of the `undefined` rule.

Conceivably we could invent a new syntax for `_L_[_i_] if it's in-bounds, otherwise *undefined*`. But there are so few alg steps that could use it, I don't think it'd be worth it.

(A lot more steps could make use of 'ternary operator'. Then here you could say something like:
`1. Let _byteOffset_ be _args_[1] if _numberOfArgs_ &gt; 1, else *undefined*.`
But I'm still doubtful it'd be worth it.)